src/components/Button/Button.test.tsx

```diff
@@ -3,62 +3,78 @@ import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { Button } from './Button';
 
+const testLabel = 'button';
+
+const buttonBaseClass = 'a-btn';
+const buttonSuperClass = 'a-btn__super';
+const buttonLinkClass = 'a-btn__link';
+const buttonFullXSClass = 'a-btn__full-on-xs';
+const buttonSecondaryClass = 'a-btn__secondary';
+
 describe('<Button />', () => {
-  it('renders labels correctly', () => {
-    render(<Button label='button' />);
-    expect(screen.getByText('button')).toBeInTheDocument();
+  it('Propagates additional HTML properties to main component element', () => {
+    const testClass = 'this-is-a-test-classname';
+    const testTitle = 'test-title';
 
-    render(<Button label='wowzers' />);
-    expect(screen.getByText('wowzers')).toBeInTheDocument();
+    render(
+      <Button label={testLabel} className={testClass} title={testTitle} />
+    );
+    expect(screen.getByText(testLabel)).toBeInTheDocument();
+    expect(screen.getByText(testLabel)).toHaveClass(testClass);
+    expect(screen.getByText(testLabel)).toHaveAttribute('title', testTitle);
   });
-  it('renders multiple appearances', () => {
-    render(<Button label='button' />);
-    expect(screen.getByText('button')).toHaveClass('a-btn');
 
-    render(<Button label='secondary button' appearance='secondary' />);
-    expect(screen.getByText('secondary button')).toHaveClass(
-      'a-btn__secondary'
-    );
+  it('Renders as a link', () => {
+    render(<Button label={testLabel} asLink />);
+    expect(screen.getByText(testLabel)).toHaveClass(buttonLinkClass);
   });
-  it('renders multiple sizes', () => {
-    render(<Button label='Button' size='default' />);
-    expect(screen.getByText('Button')).toHaveClass('a-btn');
 
-    render(<Button label='super button' size='super' />);
-    expect(screen.getByText('super button')).toHaveClass('a-btn__super');
+  it('Renders labels correctly', () => {
+    render(<Button label={testLabel} />);
+    expect(screen.getByText(testLabel)).toBeInTheDocument();
+    expect(screen.getByText(testLabel).textContent).toEqual(testLabel);
 
-    render(<Button label='full width button' size='full' />);
-    expect(screen.getByText('full width button')).toHaveClass(
-      'a-btn__full-on-xs'
-    );
+    const wowLabel = 'wowzers';
+    render(<Button label={wowLabel} />);
+    expect(screen.getByText(wowLabel)).toBeInTheDocument();
+    expect(screen.getByText(wowLabel).textContent).toEqual(wowLabel);
   });
-  it('can be disabled', () => {
-    render(<Button label='normal button' />);
-    expect(screen.getByText('normal button')).toBeEnabled();
 
-    render(<Button label='disabled button' isDisabled />);
-    expect(screen.getByText('disabled button')).toBeDisabled();
```

Comment:
How would you feel about leaving this disabled test but replacing `isDisabled` with a `disabled` prop to confirm that it gets passed down to the html button?

Reply:
[That test does still exist](https://github.com/cfpb/design-stories/pull/112/files/22a06ff67ca173911c5f6713b6eb80157f9a99aa#diff-68b91b497740b3078f98243bb823ddc8a0264fe0178b965e04c081e98c58d7b3R65-R72), I guess I messed up the `git diff` because I capitalized all of the test label.